week-01/assessment-1/src/main/java/HotelApp.java

```diff
@@ -0,0 +1,343 @@
+            System.out.println("===============");
+//            System.out.println("Guest Name: " + guestName);
+//            guestName = scan6.nextLine();
+//        System.out.println("Capsule #[1-" + numCapsules + "]:  ");
+//        System.out.println("Please choose a room: ");
+            // int chosenRoom = scan3.nextInt();
+            // isRoomAvailable(chosenRoom, capsulesArray);
+            boolean validInput = false;
+            while(!validInput) {
+                System.out.println("Capsule #[1-" + numCapsules + "]:  ");
+                System.out.println("Please choose a room: ");
+                chosenRoom = scan5.nextInt();
+                //   System.out.println("Please choose a room: ");
+                if (chosenRoom <= 0 || chosenRoom >= capsulesArray.length) {
+                    System.out.println("Not in range. Try again.");
+                } else {
+                    if (capsulesArray[chosenRoom] == null) {
+                        System.out.println("Room is unoccupied. Try again!");
+                        System.out.println("Please choose a room: ");
+                        chosenRoom = scan5.nextInt();
+
+                    } else {
+                        validInput = true;
+                    }
+                }
+//             else {
+//                System.out.println("Error! Try again.");
+//                chosenRoom = scan3.nextInt();1
+//            }
+            }
+            System.out.println("Success! " + capsulesArray[chosenRoom] + " has checked out of room #" + chosenRoom + ".");
+            capsulesArray[chosenRoom] = null;
+
+        }
+
+
+
+//
+    public static boolean exit(){
+        boolean doExit = false;
+        Scanner scan4 = new Scanner(System.in);
+        System.out.println("Exit");
+        System.out.println("====");
+        System.out.println("Are you sure you want to exit?");
+        System.out.println("All data will be lost.");
+        System.out.println("Exit [y/n]: ");
+        String exit1 = scan4.nextLine();
+
+        if (exit1.equals("y")) {
+            doExit = true;
+            System.out.println("Goodbye!");
+
+        }
+
+        return doExit;
+
+    }
+
+
+
+    public static void viewRooms(String[] capsuleArray){
+        Scanner scan4 = new Scanner(System.in);
+//        Random random = new Random();
+//        capsules = capsuleCount();
+//        StringBuilder occupant = new StringBuilder();
+
+        String guestName = "";
+        System.out.println("View Guests");
+        System.out.println("============");
+        System.out.println("Capsule #[1-" + capsuleArray.length + "]:  " + " ");
+        System.out.println("Please choose a room: ");
+        int capsuleNumber = Integer.parseInt(scan4.nextLine());
+        int max = capsuleNumber + 6;
+        int min = capsuleNumber - 5;
+
+
+
+
+        for(int i = min; i < max; i++){
+            int safeIndex = i;
+
+            if(max > capsuleArray.length){
```

Comment:
Nice solution for keeping errors for occuring while displaying rooms at the beginning and end of the rooms array!